pkg/plugins/runtime/k8s/controllers/gatewayapi/gateway_controller.go

```diff
@@ -114,6 +133,12 @@ func (r *GatewayReconciler) getGatewayClass(ctx context.Context, name gatewayapi
 	return class, nil
 }
 
+func serviceTagForGateway(gateway *gatewayapi.Gateway) map[string]string {
+	return map[string]string{
+		mesh_proto.ServiceTag: fmt.Sprintf("%s_%s_gateway", gateway.Name, gateway.Namespace),
```

Comment:
I missed this in an earlier review but we should file an issue to guarantee the service tag is unique, maybe prefix the tag value with `gateways.kuma.io/`?

Reply:
Sure, though we don't do this when we generate a `%s_%s_svc` tag (which I could for consistency imagine prefixing with `k8s.kuma.io`)

Reply:
Thinking about this some more; it's likely that people will use Gateway API, but then need to target that with Kuma policy, so if we use a prefix and a simple service name, this might be easier.

We can split this into a separate issue to discuss in more detail if you prefer.

Reply:
https://github.com/kumahq/kuma/issues/3515